### PR TITLE
Fix dockerized plugin environment (many improvements)

### PR DESCRIPTION
**Pull request by strk**
Will need a few iterations

### CODE REVIEW

**Comment by strk** [comment]
I just realized the plugin is only compatible with qgis 2.x:
> This plugin is incompatible with this version of QGIS Plugin designed for QGIS 2.4 - 2.99

So, the docker image needs to be changed to use older qgis

**Comment by strk** [comment]
Ok current status within this PR is that we have a dockerized QGIS-2.14 with the NZGBplugin installed and enabled, and a top-level Makefile which can be used to start such QGIS-2.14 connected to the caller's X DISPLAY.

**Comment by strk** [comment]
Docker changed in this PR to use QGIS-2.14, and have NZGBPlugin pre-installed.
From top-level dir, just run:

> make docker-qgis-start

To have  qgis 2.14 started with NZGBplugin enabled.

The plugin does not work, but that's some additional step to take (configure it to use appropriate database etc.)

**Comment by SPlanzer** [comment]
>I just realized the plugin is only compatible with qgis 2.x:
    This plugin is incompatible with this version of QGIS Plugin designed for QGIS 2.4 - 2.99
So, the docker image needs to be changed to use older qgis

In-short:
The purpose of this piece of work is to port to 3.10. therefore it should be 3:10

Longer:
There has been some change of scope here recently with what version of QGIS tests should be written for. 

The version of QGIS we require is dependent on our approach to automated testing. Originally we discussed writing automated tests retrospectively prior to port the plugin from QGIS2 to QGIS3. However in subsequent conversations (some of which are captured in #61) we have addressed concerns over how much work it as going to be to write QGIS2 automated tests and then have to port these test straight away to QGIS3 (this will involve porting test to python2>3, pyqt4>5 and QGIS2>3). It has been decided that this is too inefficient and as we have no preexisting test we will port the plugin to QGIS3 and then write automated tests. 

Therefore the docker qgis container QGIS version should be 3 (specifically 3.10 as per #3 4)

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
-  test:
+  test_schema:
 
-    runs-on: ${{ matrix.os }}
+    name: DB schema
+    runs-on: ubuntu-18.04
     strategy:
         matrix:
-            os: [ ubuntu-18.04 ]
             pg: [ '10' ]
```
Comment (SPlanzer) [inline .github/workflows/ci.yml]: Why does the plugin env require postgres?

Comment (strk) [reply .github/workflows/ci.yml]: it became a single workflow now. The workflow contains 2 jobs. Only the job for testing the DB schema requires postgres/postgis.

**Comment by strk** [comment]
This is now ready to be merged. Note it still uses QGIS 2 for now, but can be improved later.
What this PR brings in, is the ability to run the plugin for testing very very easly.
Just run `make docker-qgis-start` from the top-level dir and you should get a QGIS instance with the NZGBplugin installed and enabled and already configured to connect to a scratch DB in its own docker container (qgis also runs in a docker container).

It helped me doing some improvements here and there to the plugin (see open PRs).

**Pull request merged**